provider/cloudflare/cloudflare.go

```diff
@@ -394,34 +385,37 @@ func (p *CloudFlareProvider) submitChanges(ctx context.Context, changes []*cloud
 			}
 
 			resourceContainer := cloudflare.ZoneIdentifier(zoneID)
+			records, err := p.listDNSRecordsWithAutoPagination(ctx, zoneID)
+			if err != nil {
+				return fmt.Errorf("could not fetch records from zone, %v", err)
+			}
+			chs, chErr := p.listCustomHostnamesWithPagination(ctx, zoneID)
+			if chErr != nil {
+				return fmt.Errorf("could not fetch custom hostnames from zone, %v", chErr)
```

Comment:
not covered with unit tests

![Screenshot 2025-03-06 at 07 46 10](https://github.com/user-attachments/assets/4153d24b-6f76-4e2a-baf6-ca10799322b3)

Reply:
covered